scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +514,112 @@ def _logcdf(self, k, M, n, N):
 hypergeom = hypergeom_gen(name='hypergeom')
 
 
+class nhypergeom_gen(rv_discrete):
+    r"""A negative hypergeometric discrete random variable.
+
+    The negative hypergeometric distribution models the probability
+    of :math:`k` successes in a sample of exactly :math:`r` failures
+    taken without replacement where :math:`N` are the total number
+    of elements and :math:`K` are defined to be successes.
+
+    It is equivalent to observing :math:`k` successes in :math:`k+r-1`
+    samples with :math:`k+r`'th sample being a failure. The former
+    can be modelled as a hypergeometric distribution. Refer
+    `hypergeom` distribution for more information.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The symbols used to denote the shape parameters (`N`, `K`, and `r`) are not
+    universally accepted. See the Examples for a clarification of the
+    definitions used here.
+
+    The probability mass function is defined as,
+
+    .. math:: f(k; N, K, r) = \frac{{{k+r-1}\choose{k}}{{N-r-k}\choose{K-k}}}
+                                   {{N \choose K}}
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    See Also
+    --------
+    hypergeom
+
+    """
+    def _get_support(self, N, K, r):
+        return 0, K
+
+    def _argcheck(self, N, K, r):
+        cond = (N >= 0)
+        cond &= (K >= 0) & (K <= N)
+        cond &= (r >= 0) & (r <= N-K)
+        return cond
+
+    def _logpmf(self, k, N, K, r):
```

Comment:
Reference, please.

Reply:
I have added reference in the class documentation. Should the references to methods too? (don't see much use of that as it won't be cited anywhere anyways)

Reply:
If you got this formula from Wikipedia, it would be good to reference [1] in the comments here. But I don't see this formula on Wikipedia, so I would suggest:

- adding a reference [2] to the docstring (wherever you found this formula)
- mentioning in the Notes section of the docstring that the pmf is evaluated using the formula from [2] rather than the definition from [1] for better numerical accuracy
- It would help future maintainers to refer to [2] right here as a comment, too, perhaps including more specific information - e.g. page / equation number. It might seem redundant, but it's not really - what you write here is for a different audience. Maintainers of this code need more detail than users of the function.

Reply:
Oh, ok! Got you. But the formula used in the PMF function is not taken from anywhere. Instead, it is just using beta function to evaluate factorial which is faster and is more stable numerically. So, in this case, there is no reference.

Reply:
I got the formulas used in `_stats` method from a reference though. I will add that reference!